src/server/tcp.rs

```diff
@@ -43,7 +43,11 @@ impl Server {
         S::Instance: Send + Sync + 'static,
     {
         let service = Arc::new(service);
-        let listener = TcpListener::bind(self.socket_addr).await?;
+        let socket = Socket::new(Domain::IPV4, Type::STREAM, None)?;
```

Comment:
Why is the `protocol` parameter set to `None`? What would happen if we use `Protocol::TCP`?